src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -0,0 +1,170 @@
+     * A permission with this rank can be granted to anybody including the
+     * untrusted guests. For most servers there is no difference between
+     * {@link #RANK_GUEST} and {@link #RANK_USER}.
+     *
+     * <p>Examples can be chat, MotD receive.</p>
+     */
+    static final String RANK_GUEST = "guest";
+    /**
+     * A permission with this rank can be granted to anybody that should have
+     * default user access. For most servers there is no difference between
+     * {@link #RANK_GUEST} and {@link #RANK_USER}.
+     *
+     * <p>Examples can be claim city plot, access to games and public warps.</p>
+     */
+    static final String RANK_USER = "user";
+    /**
+     * A permission with this rank should only be granted to known server staff.
+     * The permission's impact on the server is quite low.
+     *
+     * <p>Examples can be give items, teleport other players, access to support
+     * commands, globally mute other players.</p>
+     */
+    static final String RANK_STAFF = "staff";
+    /**
+     * A permission with this rank should only be granted to fully trusted
+     * server administrators. The permission's impact on the server may be high.
+     *
+     * <p>Examples can be create/delete worlds, grant permissions, restart
+     * server.</p>
+     */
+    static final String RANK_ADMIN = "admin";
+    /**
+     * A permission with this rank should normally not granted to anybody. This
+     * usually applies to debug permissions.
+     */
+    static final String RANK_NOBODY = "nobody";
+
+    /**
+     * Gets the permission id this description belongs too. May include
+     * placeholders {@code "<SomeParameter>"} such as in
+     * {@code "pluginId.give.item.<ItemType>"}.
+     *
+     * @return The permission id
+     */
+    String getId();
+
+    /**
+     * Gets a short description of the linked permission. May include a link to
+     * a more detailed description on the plugin's web page.
+     *
+     * @return A short description of the linked permission
+     */
+    Text getDescription();
+
+    /**
+     * Gets the human readable <b>suggested</b> rank the players should have
+     * before they should have access to this permission.
+     *
+     * <p><b>Note:</b> This is a suggestion for server owners and does not have
+     * any impact on the results of the permission checks.</p>
+     *
+     * @return The suggested player rank the permission should be granted to
+     */
+    Optional<String> getSuggestedRank();
+
+    /**
+     * Gets the owning plugin the permission belongs to.
+     *
+     * @return The owning plugin the permission belongs to
+     */
+    PluginContainer getOwner();
+
+    /**
+     * A builder for permission descriptions.
+     */
+    interface Builder {
+
+        /**
+         * Sets the permission id for the description this builder creates. May
+         * include placeholders {@code "<SomeParameter>"} such as in
+         * {@code "pluginId.give.item.<ItemType>"}.
```

Comment:
Not sure about this. If it is going to be implemented like, there should be a stricter/more formal grammar. Take a look at EBNF for inspiration.

Also would be good to spend some thought on what uses these parameters will have in-plugin. I think plugins would have to end up stripping the parameters from permissions when presenting them to a user (autocomplete for example). Might be good to provide them as help or as some sort of fill-in-the-blanks though.